packages/fxa-settings/src/pages/ResetPassword/ConfirmResetPassword/index.tsx

```diff
@@ -2,37 +2,74 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React from 'react';
-import { RouteComponentProps } from '@reach/router';
-import { usePageViewEvent } from '../../../lib/metrics';
+import React, { useCallback, useState } from 'react';
+import { RouteComponentProps, useLocation, useNavigate } from '@reach/router';
+import Banner, { BannerType } from '../../../components/Banner';
 
 import LinkRememberPassword from '../../../components/LinkRememberPassword';
 import ConfirmWithLink, {
   ConfirmWithLinkPageStrings,
 } from '../../../components/ConfirmWithLink';
 import { REACT_ENTRYPOINT } from '../../../constants';
+import AppLayout from '../../../components/AppLayout';
+import { useAccount, useInterval } from '../../../models';
+import { FtlMsg } from 'fxa-react/dist/lib/utils';
+import { logPageViewEvent } from '../../../lib/metrics';
 
 export const viewName = 'confirm-reset-password';
 
 export type ConfirmResetPasswordProps = {
   email: string;
-  forceAuth?: boolean;
-  canSignIn?: boolean;
+  passwordForgotToken: string;
 };
 
-const ConfirmResetPassword = ({
-  email,
-  forceAuth,
-  canSignIn,
-}: ConfirmResetPasswordProps & RouteComponentProps) => {
-  usePageViewEvent(viewName, REACT_ENTRYPOINT);
+const POLLING_INTERVAL_MS = 2000;
 
-  // TODO check for passwordResetToken (confirms reset password initiated)
-  // TODO redirect to reset_password if !passwordResetToken
+const ConfirmResetPassword = (_: RouteComponentProps) => {
+  logPageViewEvent(viewName, REACT_ENTRYPOINT);
 
-  const resendHandler = () => {
-    // TODO resend code
-    // TODO logViewEvent metric
+  const navigate = useNavigate();
+  let { state = {} } = useLocation();
+  const account = useAccount();
+  const [passwordResetResend, setPasswordResetResend] = useState(false);
+  const [isPolling, setIsPolling] = useState<number | null>(
+    POLLING_INTERVAL_MS
+  );
+
+  const navigateToPasswordReset = useCallback(() => {
+    navigate('reset_password?showReactApp=true', { replace: true });
+  }, [navigate]);
+
+  if (!state) {
+    state = {};
+  }
+
+  let { email, passwordForgotToken } = state as ConfirmResetPasswordProps;
+
+  if (!email || !passwordForgotToken) {
+    navigateToPasswordReset();
+  }
+
+  useInterval(async () => {
+    try {
+      // A bit unconventional but this endpoint will throw an invalid token error
```

Comment:
It's definitely good to note that the resetPasswordStatus function behaves this way. I found this comment a little confusing at first. Initially I thought it was saying that the try/catch here was designed to catch the invalid token error, when in fact it is the try catch inside the resetPasswordStatus function that catches and returns true in the event of an invalid token error. So maybe the comment should note the function handles the error and true/false?


Just a side note, but this does feel pretty unconventional. Is there a reason we can’t have a resetPasswordStatus resolver that just provides true/false for us?